File: testdata/jennies/builders/panel_builders/PHPBuilder/src/Panelbuilder/PanelBuilder.php
<?php

namespace Grafana\Foundation\Panelbuilder;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\Panelbuilder\Panel>
 */
class PanelBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\Panelbuilder\Panel $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\Panelbuilder\Panel();
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\Panelbuilder\Panel
     */
    public function build()
    {
        return $this->internal;
    }

    public function onlyFromThisDashboard(bool $onlyFromThisDashboard): static
    {
        $this->internal->onlyFromThisDashboard = $onlyFromThisDashboard;
    
        return $this;
    }

    public function onlyInTimeRange(bool $onlyInTimeRange): static
    {
        $this->internal->onlyInTimeRange = $onlyInTimeRange;
    
        return $this;
    }

    /**
     * @param array<string> $tags
     */
    public function tags(array $tags): static
    {
        $this->internal->tags = $tags;
    
        return $this;
    }

    public function limit(int $limit): static
    {
        $this->internal->limit = $limit;
    
        return $this;
    }

    public function showUser(bool $showUser): static
    {
        $this->internal->showUser = $showUser;
    
        return $this;
    }

    public function showTime(bool $showTime): static
    {
        $this->internal->showTime = $showTime;
    
        return $this;
    }

    public function showTags(bool $showTags): static
    {
        $this->internal->showTags = $showTags;
    
        return $this;
    }

    public function navigateToPanel(bool $navigateToPanel): static
    {
        $this->internal->navigateToPanel = $navigateToPanel;
    
        return $this;
    }

    public function navigateBefore(string $navigateBefore): static
    {
        $this->internal->navigateBefore = $navigateBefore;
    
        return $this;
    }

    public function navigateAfter(string $navigateAfter): static
    {
        $this->internal->navigateAfter = $navigateAfter;
    
        return $this;
    }

}

File: testdata/jennies/builders/panel_builders/PythonBuilder/builders/panelbuilder.py
import typing
from ..cog import builder as cogbuilder
from ..models import panelbuilder


class Panel(cogbuilder.Builder[panelbuilder.Panel]):
    _internal: panelbuilder.Panel

    def __init__(self):
        self._internal = panelbuilder.Panel()

    def build(self) -> panelbuilder.Panel:
        """
        Builds the object.
        """
        return self._internal    
    
    def only_from_this_dashboard(self, only_from_this_dashboard: bool) -> typing.Self:    
        self._internal.only_from_this_dashboard = only_from_this_dashboard
    
        return self
    
    def only_in_time_range(self, only_in_time_range: bool) -> typing.Self:    
        self._internal.only_in_time_range = only_in_time_range
    
        return self
    
    def tags(self, tags: list[str]) -> typing.Self:    
        self._internal.tags = tags
    
        return self
    
    def limit(self, limit: int) -> typing.Self:    
        self._internal.limit = limit
    
        return self
    
    def show_user(self, show_user: bool) -> typing.Self:    
        self._internal.show_user = show_user
    
        return self
    
    def show_time(self, show_time: bool) -> typing.Self:    
        self._internal.show_time = show_time
    
        return self
    
    def show_tags(self, show_tags: bool) -> typing.Self:    
        self._internal.show_tags = show_tags
    
        return self
    
    def navigate_to_panel(self, navigate_to_panel: bool) -> typing.Self:    
        self._internal.navigate_to_panel = navigate_to_panel
    
        return self
    
    def navigate_before(self, navigate_before: str) -> typing.Self:    
        self._internal.navigate_before = navigate_before
    
        return self
    
    def navigate_after(self, navigate_after: str) -> typing.Self:    
        self._internal.navigate_after = navigate_after
    
        return self
    
